src/components/layout/header/menu-flyout.tsx

```diff
@@ -60,12 +60,65 @@ interface NavigationFlyoutProp {
   translation?: string;
   showLanguageSwitch?: boolean;
   setIsNavigationVisible: (b: boolean) => void;
+  burgerRef: React.RefObject<HTMLButtonElement>;
 }
 
 export const NavigationFlyout: React.FC<NavigationFlyoutProp> = (props) => {
+  const overlayRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const preventFocusOfNonFlyoutContent = (visible: boolean) => {
+      const focusableElements = overlayRef.current?.querySelectorAll(
+        'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])',
+      ) as NodeListOf<HTMLElement>;
+
+      const firstFocusableElement = focusableElements?.[0];
+      const lastFocusableElement =
+        focusableElements?.[focusableElements.length - 1];
+
+      const handleFocus = (event: KeyboardEvent) => {
+        if (event.key !== 'Tab') {
+          return;
+        }
+        const shiftKeyPressed = event.shiftKey;
+        const activeElement = document.activeElement;
+        const burgerElement = props.burgerRef.current;
+        if (shiftKeyPressed && activeElement === burgerElement) {
+          lastFocusableElement?.focus();
+          event.preventDefault();
+        }
+        if (!shiftKeyPressed && activeElement === burgerElement) {
+          firstFocusableElement?.focus();
+          event.preventDefault();
+        }
+        if (shiftKeyPressed && activeElement === firstFocusableElement) {
+          props.burgerRef.current?.focus();
+          event.preventDefault();
+        }
+        if (!shiftKeyPressed && activeElement === lastFocusableElement) {
+          props.burgerRef.current?.focus();
+          event.preventDefault();
+        }
```

Comment:
Can you please add the new constant here? 

```suggestion
        if (shiftKeyPressed && activeElement === firstFocusableElement) {
          burgerElement?.focus();
          event.preventDefault();
        }
        if (!shiftKeyPressed && activeElement === lastFocusableElement) {
          burgerElement?.focus();
          event.preventDefault();
        }
```